plugins/sim/cdsimcontroller.cpp

```diff
@@ -402,28 +491,45 @@ void CDSimController::ensureSimContactsPresent()
 
 void CDSimController::voicemailConfigurationChanged()
 {
-    if (!m_voicemailConf || !m_simManager.present()) {
+    ModemData *modem = 0;
+    if (MGConfItem *voicemailConf = qobject_cast<MGConfItem *>(sender())) {
```

Comment:
Yeah, I think that would be a reasonable clean up, but first I'd rather have as direct a translation of the existing code as possible.  Testing is likely to be somewhat inadequate, so I'd like to verify it by inspection as much as possible.